docs/reactivity/reactive-components.md

```diff
@@ -0,0 +1,99 @@
+  @property() sku?: string;
+  protected productController = new ProductController(this);
+  protected product$ = this.productController.get();
+}
+```
+
+The `ProductController`resolves the product qualifier (SKU) from the context and returns an observable from the `ProductService`. It will also take a component property (`sku`) into account, in case the SKU is provided statically to the component. This might be useful in custom development or in case of demonstrating the component (e.g. Storybook).
+
+In these code snippet, an observable is assigned to the local `product$` field. This will not do anything, since you need to _subscribe_ to an observable. There's a convenient approach in Oryx to subscribe using a decorator:
+
+```ts
+export class ProductPriceComponent {
+  @subscribe()
+  protected product$ = this.productService.get({ sku: '123' }).pipe(
+    tap((product) => {
+      //do stuff here
+    })
+  );
+}
+```
+
+The `subscribe` decorator will subscribe to the observable, but also unsubscribe when the component is destroyed. This will ensure that there's no leaking memory in the application.
+
+Another missing piece here is how the observed data can be used and rendered inside the component. This is described in the next section.
+
+## Update the DOM
+
+While the user navigates through pages in a single page application experience, it's crucial for components to be updated accordingly. This section will explain how data updates are propagated throughout the user interface in Oryx.
+
+RxJS conveniently operates on data streams and updates them in memory, but it won't synchronize this to the UI automatically. Each JavaScript framework ships its own opinionated method to update the DOM. The method of choice contributes significantly to the performance and user experience of the application.
+
+The components provided in the Oryx libraries are build with Lit. Lit provides a highly efficient system to only synchronizes the minimum required updates to the DOM. When updates are loaded asynchronous, we need, however, to update the UI whenever new data is emitted. To make this as transparent as possible, a decorator (`@asyncState()`) is provided that can be used in the UI components. The decorator uses the `AsyncStateController` under the hood, which will request updates to the view whenever needed.
+
+The following example shows the use of the `asyncState` decorator. The decorator subscribes to the assigned observable and requests an update to the component whenever a needed. This means that as a component developer you do not need to worry about how the reactive system works under the hood.
+
+Oryx components are build in TypeScript, which is why we need to ensure type safety. The original type of the assigned observable needs to be adjusted. It's not possible to resolve correct TypeScript type from the observable by just using decorator, which is why the `valueType` function is offered to project the observed type.
```

Comment:
Is it the reader or Spryker who needs to ensure type safety?

Reply:
We are refactoring asyncState and valueType, so let's wait a little as this text will likely become obsolete.